Library/Homebrew/dev-cmd/pr-pull.rb

```diff
@@ -239,6 +416,7 @@ def pr_pull
         cd dir do
           original_commit = Utils.popen_read("git", "-C", tap.path, "rev-parse", "HEAD").chomp
           cherry_pick_pr!(pr, path: tap.path, args: args)
+          autosquash!(original_commit, path: tap.path, args: args) if args.autosquash?
           signoff!(pr, tap: tap, args: args) unless args.clean?
```

Comment:
I wonder if these three functions could all split off into other files with unit tests. Just aware that this file is getting pretty long, only likely to get longer and has essentially no testing.

Reply:
Yeah, I'm thinking of maybe moving into `utils/git.rb`. Have split some stuff off into their own functions and thinking about how I can do tests.